go-controller/pkg/node/gateway_shared_intf_linux.go

```diff
@@ -38,7 +38,15 @@ func setupLocalNodeAccessBridge(nodeName string, subnets []*net.IPNet) error {
 		return err
 	}
 
-	macAddress := util.IPAddrToHWAddr(net.ParseIP(util.V4NodeLocalNatSubnetNextHop)).String()
+	var macAddress string
+	if config.IPv4Mode && config.IPv6Mode {
+		macAddress = util.IPAddrToHWAddr(net.ParseIP(util.V4NodeLocalNatSubnetNextHop)).String()
+	} else if config.IPv4Mode {
+		macAddress = util.IPAddrToHWAddr(net.ParseIP(util.V4NodeLocalNatSubnetNextHop)).String()
```

Comment:
you don't need the first case; if `config.IPv4Mode` is set, you use the IPv4 MAC, regardless of whether `config.IPv6Mode` is set

Reply:
yeah I thought about that, but someone added this exact structure to where we set the mac binding, so I thought you probably added it to be more explicit

Reply:
simplified it in latest commit